public/containers/FlushIndexModal/FlushIndexModal.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useContext, useState, useEffect } from "react";
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSpacer,
+} from "@elastic/eui";
+import { CoreStart } from "opensearch-dashboards/public";
+import { CoreServicesContext } from "../../components/core_services";
+import { getErrorMessage } from "../../utils/helpers";
+import { ServicesContext } from "../../services";
+import { indexBlockedPredicate, aliasBlockedPredicate, dataStreamBlockedPredicate, filterBlockedItems } from "../../utils/helpers";
+import { IndexOpBlocksType } from "../../utils/constants";
+import { CatIndex, DataStream } from "../../../server/models/interfaces";
+import { IAlias } from "../../pages/Aliases/interface";
+
+type FlushTarget = "indices" | "data stream" | "alias";
+
+const flushAllMessage = "all open indexes will be flushed.";
+
+const messageMap: Record<FlushTarget, string> = {
+  indices: "The following indexes will be flushed:",
+  "data stream": "The following data streams will be flushed:",
+  alias: "The following aliases will be flushed:",
+};
+
+const blockedMessageMap: Record<FlushTarget, string> = {
+  indices: "The following indexes will not be flushed because they are closed:",
+  "data stream": "The following data streams will not be flushed because one or more backing indexes are closed:",
+  alias: "The following aliases will not be flushed because one or more indexes are closed:",
+};
+
+export interface FlushIndexModalProps<T> {
+  selectedItems: T[];
+  visible: boolean;
+  flushTarget: FlushTarget;
+  onClose: () => void;
+}
+
+export default function FlushIndexModal<T>(props: FlushIndexModalProps<T>) {
+  const { onClose, flushTarget, visible, selectedItems } = props;
+  const services = useContext(ServicesContext);
+  const coreServices = useContext(CoreServicesContext) as CoreStart;
+  const flushAll = !selectedItems.length && flushTarget === "indices";
+
+  const [unBlockedItems, setUnBlockedItems] = useState([] as string[]);
+  const [blockedItems, setBlockedItems] = useState([] as string[]);
+  const onFlushConfirm = useCallback(async () => {
+    if (!services) {
+      coreServices.notifications.toasts.addDanger("Something is wrong in ServiceContext");
+      onClose();
+      return;
+    }
+    const indexPayload = unBlockedItems.join(",");
+    const result = await services.commonService.apiCaller({
+      endpoint: "indices.flush",
+      data: {
+        index: indexPayload,
+      },
+    });
+    if (result && result.ok) {
+      const flushedItems = flushAll ? "all open indexes" : `[${indexPayload}]`;
+      coreServices.notifications.toasts.addSuccess(`Flush ${flushedItems} successfully`);
```

Comment:
join(", "), add additional space to make the message truncatable.

Reply:
will change it